src/main/java/net/fabricmc/mappingio/MappingReader.java

```diff
@@ -43,6 +43,26 @@ public static MappingFormat detectFormat(Path file) throws IOException {
 		if (Files.isDirectory(file)) {
 			return MappingFormat.ENIGMA_DIR;
 		} else {
+			String fileName = file.getFileName().toString();
+			int dotIdx = fileName.lastIndexOf('.');
+
+			if (dotIdx >= 0) {
+				String ext = fileName.substring(dotIdx + 1);
+
+				switch (ext) {
+				case "mapping": // Standard name
+				case "mappings": // Sometimes used for standalone (non-dir) files
+				case "enigma": // Erroneously used by Lorenz <0.6
```

Comment:
Why support it then?

Reply:
Why not? It's unique enough to not cause any issues

Reply:
Maybe add an overload to opt out of this ext detection?

Reply:
Done